File: JavaScript/Manzano/Pagina46l.js
// Programa que efetua a leitura de valores positivos inteiros até que um valor negativo seja informado.
// Ao final, devem ser apresentados o maior e o menor valores informados pelo usuário.

const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let numero = 0;
let maximo = 0;
let minimo = 0;

function lerNumero() {
  rl.question("Digite um número: ", function (input) {
    numero = parseInt(input);

    if (numero >= 0) {
      maximo = numero;
      minimo = numero;

      lerNumero();
    } else {
      console.log("O maior número escolhido é: " + maximo);
      console.log("O menor número escolhido é: " + minimo);
      rl.close();
    }
  });
}

lerNumero();

File: Java/src/Manzano/Pagina46l.java
package Manzano;
//Programa que efetua a leitura de valores positivos inteiros até que um valor negativo
//seja informado. Ao final devem ser apresentados o maior e o menor valores informados pelo
//usuário.

import java.util.Scanner;

public class Pagina46l {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);

        System.out.println("Digite um número: ");
        int numero = scanner.nextInt();
        int maximo = 0;
        int minimo = 0;
        if (numero >= 0) {
            maximo = numero;
            minimo = numero;
            while (numero >= 0) {
                System.out.println("Digite um número; ");
                numero = scanner.nextInt();
                if (numero > maximo) {
                    maximo = numero;
                } else if (numero < minimo && numero>=0) {
                    minimo = numero;
                }
            }
        }
        System.out.println("O maior número escolhido é o: " + maximo);
        System.out.println("O menor número escolhido é o: " + minimo);


    }
}